department-of-veterans-affairs/caseflow
# Removing need to choose payee_code

Connects #7524

### Description
Added a new method in the `SelectClaimant` component which only shows the PayeeCode dropdown if it is needed (i.e. the form type is either supplement claim, or higher level review and the benefit type is either pension or compensation).

Also removed the backend's inferring of payee code, and now always send it from the frontend. i.e. we no longer guess that the payee code is "00" if unspecified by the frontend, and always specify a payee code of "00" when the claimant is the Veteran.

### Testing Plan
1. Go through the supplement claim or higher level review paths.
   1. Once when choosing pension/compensation with and without non-veteran claimants
      1. Ensure that with a veteran claimant the payee code is recorded as 00
      1. Ensure that with a non veteran claimant the payee code is whatever is specified on the frontend.
   1. Once when choosing fiduciary
      1. Ensure that the payee code is nil
1. Go through the appeal path
    1. Ensure the payee code is nil
